pkg/authn/types.go

```diff
@@ -0,0 +1,56 @@
+package authn
+
+import (
+	"net/http"
+
+	"github.com/bacalhau-project/bacalhau/pkg/lib/provider"
+)
+
+// The rule that authentication policies must implement. If the rule returns a
+// token string, the authentication method has succeeded and passed policy, and
+// the token string will be passed to future API calls. If the rule returns
```

Comment:
> the token string will be passed to future API calls

Is this always via Authorization header with a "Bearer <token>"?

Reply:
Yes, but that's more of an authz concern. That's how it's currently set out in the draft docs.